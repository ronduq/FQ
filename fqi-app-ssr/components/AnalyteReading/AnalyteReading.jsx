import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash.get';
import classNames from 'classnames';
import s from './AnalyteReading.scss';

const AnalyteReading = ({
  analyteId,
  produce,
  produceInfo,
  usdaTexts,
  content,
  retailerLabel
}) => {
  let label;
  let score;
  let recentScan;
  let usda;
  let status;
  let uom;

  Object.keys(get(produceInfo, ['info','analytes'], {})).map((analyte) => {
    const item = produceInfo.info.analytes[analyte];
    if(item.id === analyteId){
      label = item.label;
      score = item.score;
      recentScan = item.recentScan;
      usda = item.usda;
      status = item.status;
      uom = item.uom;
    }
  })

  const createAnalyteHtml = () => {
    return {__html: `${produce[analyteId]}`};
  }

  let atUsdaText = usdaTexts.green_text;
  let usdaTitle = usdaTexts.green_title;
  switch(status) {
    case 'Yellow':
      atUsdaText = usdaTexts.amber_text;
      usdaTitle = usdaTexts.amber_title;
    break;
    case 'Red':
      atUsdaText = usdaTexts.red_text;
      usdaTitle = usdaTexts.red_title;
    break;
  }

  const atTextReplaced = atUsdaText
    .replace("$analyte$", label)
    .replace("$produce$", produce.label)
    .replace("$retailer$", retailerLabel);
    
  return ( 
    <div className={s.container}>

      <ul className={s.readingMetrics}>
        <li><h3>{label} {content.reading}</h3></li>
        <li className={s.metric}><span className={s.metricTitle}>{content.score}</span> <span className={s.metricText}>{score}</span><span>{content.score_unit}</span></li>
        <li className={s.metric}><span className={s.metricTitle}>{content.scan}</span> <span className={s.metricText}>{recentScan}</span><span className={s.metricSuffix}>{uom}</span></li>
        <li className={s.metric}><span className={s.metricTitle}>{content.usda}</span> <span className={s.metricText}>{usda}</span><span className={s.metricSuffix}>{uom}</span></li>
      </ul>

      <div className={classNames(s.status,{[s.red]: status === 'Red', [s.amber]: status === 'Yellow'})}>
        <div className={s.statusLabel}>
          {usdaTitle}
        </div>
        <div className={s.statusText}>
          {atTextReplaced}
        </div>
      </div>

      <div>
        <h3>{produce.label} &amp; {label}</h3>
        {/* function createMarkup() { return {__html: 'First &middot; Second'}; }; */}
        <div className={s.analyteText} dangerouslySetInnerHTML={createAnalyteHtml()}></div> 
        
      </div>
    </div>
       
  );
};

AnalyteReading.propTypes = {
  analyteId: PropTypes.string,
  produce: PropTypes.shape(),
  produceInfo: PropTypes.shape(), 
  usdaTexts: PropTypes.shape(), 
  content: PropTypes.shape(), 
  retailerLabel: PropTypes.string,
};

AnalyteReading.defaultProps = {
  analyteId: '',
  produce: {},
  produceInfo: {},
  usdaTexts: {},
  content: {},
}

export default AnalyteReading;
