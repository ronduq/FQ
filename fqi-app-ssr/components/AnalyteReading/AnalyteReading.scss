@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';


.container {
  padding: 50px 30px;

  @include respond-to('mobile') {
    padding: 10px;  
  }
}

.readingMetrics {
  @include font-family('proxima-bold');
  @include font-size('large');

  list-style-type: none;
  display: flex;
  border-bottom: 1px solid $colour-grey-medium;
  padding-bottom: 20px;
  justify-content: space-between;

  @include respond-to('mobile') {
    flex-direction: column;
  }

  li {
    @include respond-to('mobile') {
      flex: 1 1 100%;
      text-align: left;
    }
  }
}

.metric {
  text-align: center;
  position: relative;
}

.metricTitle {
  @include font-family('proxima-light');
  @include font-size('xsmall');
  text-transform: uppercase;
  transform: translateY(-2px);
  display: inline-block;
  @include respond-to('mobile') {
    @include font-size('medium');
    text-transform: capitalize;
  }
}

.metricText {
  @include respond-to('mobile') {
    position: absolute;
    right: 35px;
  }
}

.metricSuffix {
  @include font-size('small');
  @include respond-to('mobile') {
    position: absolute;
    right: 0;
    bottom: 5px;
    width: 30px;
  }
}

.status {
  margin: 40px 0;
  padding: 20px;
  border: 1px solid $colour-green;
  background: rgba(28,172,105,0.05);
  display: flex;
  align-items: center;
  justify-content: center;

  @include respond-to('mobile') {
    flex-direction: column;
  }

  &.red {
    background: rgba(255,88,83,0.05);
    border: 1px solid $colour-red;
  }

  &.amber {
    background:rgba(248,162,55,0.05);
    border: 1px solid rgba(248,162,55,1);
  }
}

.statusLabel {
  background: $colour-green;
  color: $white;
  border-radius: 15px;
  text-transform: uppercase;
  @include font-family('proxima-bold');
  display:inline-block;
  min-width: 100px;
  padding: 0 10px;
  min-height: 28px;
  text-align: center;
  flex: 1 1 0;

  @include respond-to('mobile') {
    flex: 1 0 100%;
  }

  .red &{
    background: $colour-red;
  }

  .amber &{
    background: rgba(248,162,55,1);
  }
}

.statusText {
  padding: 0 30px;
  flex: 5 1 0;
  @include respond-to('mobile') {
    flex: 1 0 100%;
    padding: 0;
  }
}

.analyteText {
  ul {
    list-style: disc inside;
    text-indent: 10px;
  }
  strong {
    @include font-family('proxima-bold');
  }
  img {
    max-width: 100%;
  }
  em {
    font-style: italic;
  }
}