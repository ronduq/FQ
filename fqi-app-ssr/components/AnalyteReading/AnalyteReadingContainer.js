import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import AnalyteReading from './AnalyteReading';
import { 
  getUsdaContent,
  getProduceInfo,
  getProduceProfileContent,
  getSelectedRetailerLabel
} from '../../selectors/index';

const mapStateToProps = state => {
  return {
    usdaTexts: getUsdaContent(state),
    produceInfo: getProduceInfo(state),
    content: getProduceProfileContent(state),
    retailerLabel: getSelectedRetailerLabel(state)
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AnalyteReading);