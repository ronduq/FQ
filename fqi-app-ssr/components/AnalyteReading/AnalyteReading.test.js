/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import AnalyteReading from './AnalyteReading'

describe('With Snapshot Testing', () => {
  const handleClick = jest.fn();

  const props = {
    produceInfo: {
      info: {
        analytes: [ 
        {score: 99.75, recentScan: 113, usda: 73, status: "Green", id: "sucrose"},
        {score: 98, recentScan: 123, usda: 70, status: "Green", id: "fructose"},
        {score: 92.5, recentScan: 119, usda: 77, status: "Green", id: "glucose"},
        {score: 65.25, recentScan: 46, usda: 69, status: "Red", id: "vitaminc"},
        {score: 94, recentScan: 117, usda: 70, status: "Green", id: "water"},
        ]
      },
      labels: [
        {label: "Water", id: "water"},
        {label: "Glucose", id: "glucose"},
        {label: "Sucrose", id: "sucrose"},
        {label: "Fructose", id: "fructose"},
        {label: "Citric Acid", id: "citric"},
      ]
    }, 
  }
  it('AnalyteReading snapshot"', () => {
    const component = renderer.create(<AnalyteReading  {...props}/>)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });
})