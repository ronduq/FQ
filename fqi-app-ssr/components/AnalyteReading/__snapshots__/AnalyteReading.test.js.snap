// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing AnalyteReading snapshot" 1`] = `
<div
  className="container"
>
  <ul
    className="readingMetrics"
  >
    <li>
      <h3>
         
      </h3>
    </li>
    <li
      className="metric"
    >
      <span
        className="metricTitle"
      />
       
      <span
        className="metricText"
      />
      <span
        className="metricSuffix"
      />
    </li>
    <li
      className="metric"
    >
      <span
        className="metricTitle"
      />
       
      <span
        className="metricText"
      />
      <span
        className="metricSuffix"
      />
    </li>
    <li
      className="metric"
    >
      <span
        className="metricTitle"
      />
       
      <span
        className="metricText"
      />
      <span
        className="metricSuffix"
      />
    </li>
  </ul>
  <div
    className="status"
  >
    <div
      className="statusLabel"
    />
    <div
      className="statusText"
    >
       
       
       
    </div>
  </div>
  <div>
    <h3>
       & 
    </h3>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "undefined",
        }
      }
    />
  </div>
</div>
`;
